### PR TITLE
add feature flag to add minimum score threshold

### PR DESCRIPTION
**Pull request by snuffysasa**
pretty simple, not too much to say.  code is pretty self explanatory

### CODE REVIEW

**Review by simonlindholm — COMMENTED** [review]
It's more of a max score than a min score, isn't it? Or since min/max seems a bit ambiguous, perhaps a threshold, rather, or "only if below". Would be neat to fold better-only into this but I don't know how the cmdline flag UX would look for that

**Comment by snuffysasa** [comment]
I would call it a min score, not a max score...  
but --only-if-below could be more clear yea.   

yea it's basically the same thing as --better-only... if the user knows what the base_score is going to be.

but they could run it once to see, then run again with the flag.  Or we could just leave them as separate options.

or it you put the value to 0 it behaves like --better-only ?

File: src/main.py [hunk 1]
```diff
@@ -55,6 +55,7 @@ class Options:
     abort_exceptions: bool = False
     better_only: bool = False
     best_only: bool = False
+    score_threshold: int = 1
```
Comment (simonlindholm) [inline src/main.py:58]: this is an Optional[int], and the default should probably be None

File: src/permuter.py [hunk 2]
```diff
@@ -118,6 +119,7 @@ def __init__(
         self._show_errors = show_errors
         self._best_only = best_only
         self._better_only = better_only
+        self.score_threshold = score_threshold
```
Comment (simonlindholm) [inline src/permuter.py:122]: make this member private, just like the other ones

File: src/main.py [hunk 3]
```diff
@@ -642,6 +644,12 @@ def main() -> None:
         action="store_true",
         help="Removes all text above the target function in the output source files",
     )
+    parser.add_argument(
+        "--only-if-below",
+        dest="score_threshold",
+        type=int,
+        help="The minimum diff score for reporting",
```
Comment (simonlindholm) [inline src/main.py:651]: this help text is confusing

File: src/main.py (hunk 1 above)
Comment (snuffysasa) [reply src/main.py:58]: thought I had it as optional int,  will update

File: src/permuter.py (hunk 2 above)
Comment (snuffysasa) [reply src/permuter.py:122]: fixed, yea I had it private, but then when I did the rename, I accidentally changed it

File: src/main.py (hunk 3 above)
Comment (snuffysasa) [reply src/main.py:651]: updated

File: src/main.py [hunk 4]
```diff
@@ -648,7 +648,7 @@ def main() -> None:
         "--only-if-below",
         dest="score_threshold",
         type=int,
-        help="The minimum diff score for reporting",
+        help="Only report scores better(lower) than this value",
```
Comment (simonlindholm) [inline src/main.py:651]:
```suggestion
        help="Only report scores better (lower) than this value",
```

**Review by simonlindholm — APPROVED** [review]

**Pull request merged**